### DISCUSSION TITLE
[4.x]: Why eagerloading automatically returns collection ?

### DISCUSSION THREAD

**Discussion by jlachancekffein** [craftcms/cms#11562] [Ideas] [answered]
### Description

I'm not sure if it makes sense to return collections for each eagerloading item automatically.

I understand the relevance of being able to obtain collections via `->collect()`.

But I don't understand the reason to automate all this for eagerloader items.

Wouldn't it be more convenient to return an array if `->one()` or `->all()` is used for the query and collection if `->collect()` is used.

If not, would it be feasible to configure via the config/general.php configuration file the expected return type for the eagerloading system. eg: `'eagerloading_return_type' => 'array|collection'`?

### Craft CMS version

4.0.0

### Installed plugins and versions

-

**Comment by brandonkelly** [comment, marked as answer]
The entire motivation for adding support for collections was so there could be a consistent way of working with eager-loaded elements and normally-queried elements – see #8513.

> Wouldn't it be more convenient to return an array if `->one()` or `->all()` is used for the query and collection if `->collect()` is used.

Convenient how? You can do all the same things with a collection that you can with an array; you just get some added benefits on top of that.
